core/rawdb/rawdb.go

```diff
@@ -0,0 +1,63 @@
+package rawdb
+
+import (
+	"encoding/json"
+
+	"github.com/ChainSafe/gossamer/common"
+	"github.com/ChainSafe/gossamer/core/types"
+	"github.com/ChainSafe/gossamer/polkadb"
+	log "github.com/ChainSafe/log15"
+)
+
+// check checks to see if there an error if so writes err + message to terminal
+func check(e error, msg string) {
+	if e != nil {
+		log.Crit(msg, "err", e)
+	}
+}
+
+// SetHeader stores a block header into the KV-store; key is headerPrefix + hash
+func SetHeader(db polkadb.Writer, header *types.BlockHeader) {
+	hash := header.Hash
+
+	// Write the encoded header
+	bh, err := json.Marshal(header)
```

Comment:
any reason we are using json?

Reply:
I suppose no particular reason other than we don't have any way to encode the header to bytes

Reply:
Did a quick google search and it seems like Gob is more efficient and quicker than JSON. Perhaps that can be a task for the future though?

https://gist.github.com/evalphobia/a2ba2636acbc112f68dcd89e8b81d349
https://stackoverflow.com/questions/41179453/difference-between-encoding-gob-and-encoding-json-in-golang

Reply:
Yeah, I initially was using the following code snippet but taking it back from bytes to type Header didn't work so well. 

```
func ToBytes(key interface{}) []byte {
	var buf bytes.Buffer
	enc := gob.NewEncoder(&buf)
	err := enc.Encode(key)
	if err != nil {
		log.Crit("error converting to bytes", "error", err)
		return nil
	}
	return buf.Bytes()
}
```